src/types.js

```diff
@@ -59,9 +63,70 @@ export type SpacingAliases = {
   +[spacingAlias: string]: number,
 }
 
+export type DisplayAliases = {
+  +[displayAlias: string]: string,
+}
+
 export type ConfigProviderContext = {|
   +xnReflex: {|
+    +displayAliases: DisplayAliases,
     +spacingAliases: SpacingAliases,
     +base: number,
   |},
 |}
+
+export type OneResolutionGrid = {
+  align?: AlignGrid,
+  children?: React.Node,
+  className?: string,
+  dev?: Dev,
+  justify?: Justify,
+  margin?: Spacing,
+  marginBottom?: SpacingValues,
+  marginLeft?: SpacingValues,
+  marginRight?: SpacingValues,
+  marginTop?: SpacingValues,
+  padding?: Spacing,
+  paddingBottom?: SpacingValues,
+  paddingLeft?: SpacingValues,
+  paddingRight?: SpacingValues,
+  paddingTop?: SpacingValues,
+  show?: DisplayValues,
+  size?: Size,
+  style?: { [string]: string | number },
+}
+
+export const resolutionProperties = [
+  'align',
+  'justify',
+  'margin',
+  'marginBottom',
+  'marginLeft',
+  'marginRight',
+  'marginTop',
+  'padding',
+  'paddingBottom',
+  'paddingLeft',
+  'paddingRight',
+  'paddingTop',
+  'size',
+]
```

Comment:
These are the properties for which you can now pass either an object (with keys indicating when to apply it) or the previous format